### ISSUE TITLE
Support on no-explicit-any rule

### ISSUE THREAD

**Issue by zviwex** [rhys-vdw/ts-auto-guard#190]
Hey maintainer!
Not so much an issues as a proposal.
Where is a way to support [no-explicit-any](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-explicit-any.md) rule in type guards.
Is it something you’re interested to add support to? And if so, be default or in a dedicated flag?
I would love to contribute, but want to check with you first if it’s something you’ll want to support.

LMK

**Comment by rhys-vdw** [comment]
If I understand your question correctly: You should disable your linter for the generated files. If there are any errors in the files then they should be fixed on a case-by-case basis. Code style standards don't matter in generated code because nobody will be editing it.

**Comment by rhys-vdw** [comment]
I've added wontfix, but can remove it if there's something I'm not understanding here!

**Comment by Amatewasu** [comment]
Hi! Actually, I have the warning in the file that calls the guard (the file that calls `isFoo(obj)`) and not the generated file. But only sometimes, I don't really understand yet. I will update this issue when I get more insight.

**Comment by rhys-vdw** [comment]
Okay, well it would be more correct to use `obj: unknown` for the argument. Maybe when you get this issue again you can try switching it out and see if it resolves the issue. `unknown` is actually more correct, it just didn't exist when I made the library.

**Comment by sam-mfb** [comment]
Given the preference for the `unknown` type over `any` in situations where a type has not been narrowed, I would suggest that these typeguards should be generated in the form:
```typescript
export function isPerson(obj: unknown): obj is Person {
  const typedObj = obj as Person
  return (
    typeof typedObj === 'object' &&
    typeof typedObj.name === 'string' &&
    (typeof typedObj.age === 'undefined' || typeof typedObj.age === 'number') &&
    Array.isArray(typedObj.children) &&
    typedObj.children.every(e => isPerson(e))
  )
}
```
I believe this is more semantically correct and would avoid the linting error complained of in the original question

**Comment by sam-mfb** [comment]
PR implementing this proposal [here](https://github.com/rhys-vdw/ts-auto-guard/pull/196)

**Comment by download13** [comment]
Closed by #196
